core/src/main/java/com/homeaway/streamplatform/streamregistry/StreamRegistryApplication.java

```diff
@@ -390,4 +406,49 @@ private void registerManagedContainer(final Environment environment, ManagedKStr
             new StreamRegistryManagedContainer(managedKStreams, managedInfraManager, managedKafkaProducer);
         environment.lifecycle().manage(managedContainer);
     }
+
+    protected boolean isEventStoreKafkaTopicPresent(AdminClient adminClient, String topicName) {
+        DescribeTopicsResult describeTopicsResult = adminClient.describeTopics(Collections.singleton(topicName));
+        try {
+            TopicDescription topicDescription = describeTopicsResult.values().get(topicName).get();
+            log.debug("EventStore topic {} present. Topic Details = {}", topicName, topicDescription);
```

Comment:
This validation is a Special scenario.

if the topic is not available, then, Kafka API `adminClient.describeTopics` wraps the `UnknownTopicOrPartitionException` inside `ExecutionException`, and throw that back.

However, if the cluster is not available, then, it throws `ExecutionException` too.

So, we have to make sure we return `isEventStoreKafkaTopicPresent=false`, if the topic is not available. If the exception is caused by something else, we just return it and kill the application.